WordPress/gutenberg
# Widget Editor: Refactor header using the DocumentTools component from the editor package

Related to #57214

## What?

This PR refactors the header in the Widget Editor using the `DocumentTools` component from the `@wordpress/editor` package instead of the proprietary `DocumentTools` component.

At the same time, we want the "Show button text labels" setting to be applied to the header area as well.

### Before

Document tool buttons are 36px in size unlike other editors. Additionally, the focus outline is cut off.

![image](https://github.com/WordPress/gutenberg/assets/54422211/bd77b08e-eeb1-42f2-b9bf-9ade66b57ebf)

Mobile Layout:

![image](https://github.com/WordPress/gutenberg/assets/54422211/0135a052-a80e-4ea6-81a5-dcd5e6a1d41f)

### After

The document tools button size is 32 pixels, similar to other editors. Also, the focus outline is not cut.

![image](https://github.com/WordPress/gutenberg/assets/54422211/e449788c-2c4d-4e54-b1d7-437cc39f6794)

The Widget Editor doesn't have the preferences modal, but when you turn on "Show button text labels" in the Post Editor or the Site Editor, the buttons in the header display text instead of icons.

![image](https://github.com/WordPress/gutenberg/assets/54422211/01567a1a-ef2d-48c4-9b3d-9213ea5027eb)

Mobile Layout: 

![image](https://github.com/WordPress/gutenberg/assets/54422211/4a50614f-7a27-4d77-a151-198d65e69318)

![image](https://github.com/WordPress/gutenberg/assets/54422211/dca52346-ab2e-46f4-9392-4a066136ecaa)

## Why?

The Widget Editor header has a different UI compared to other editors. The approach taken in #57214 to unify `DocumentTools` component should also be applicable to the Widget Editor.

## How?

- Replace `DocumentTools`
- Reflect the "Show button text labels" setting
- Added deprecation to `setIsInserterOpened` and `setIsListViewOpened` actions
- Adjusting CSS to achieve the same layout as other editors

## Testing Instructions

- Inserting blocks and list views should work as before.
- Edit/Select mode and Undo/Redo should work as before.
- With a _closed_ widget area selected, clicking on the main block inserter should open that widget area.
- When you enable the Top Toolbar and narrow the browser width, a horizontal scrollbar should appear on the block toolbar.
